src/app.rs

```diff
@@ -75,8 +76,13 @@ pub async fn build_with_cert(
 }
 
 pub async fn build(config: config::Config) -> anyhow::Result<Bound> {
-    let cert_manager = identity::SecretManager::new(config.clone());
-    build_with_cert(config, cert_manager).await
+    if config.fake_ca {
+        let cert_manager = identity::mock::MockCaClient::new(Duration::from_secs(86400));
+        build_with_cert(config, cert_manager).await
+    } else {
+        let cert_manager = identity::SecretManager::new(config.clone());
+        build_with_cert(config, cert_manager).await
+    }
```

Comment:
There is probably some magic Rust syntax to simplify this to `let cert_manager = match config.fake_ca { ... }`, but I couldn't figure a way to fix the `match arms have incompatible types` error.